client/ayon_deadline/plugins/publish/global/submit_publish_job.py

```diff
@@ -419,11 +419,15 @@ def process(self, instance):
             "job": render_job or None,
             "instances": instances
         }
-        collected_version = (
-            instance.data.get("version")   # instance override version
-            or instance.context.data.get("version")   # workfile version
-        )
-        if collected_version:
+
+        # Note that a version of 0 is a valid version number,
+        # so we explicitly check for `None` value
+        # instance override version
+        collected_version = instance.data.get("version")
+        if collected_version is None:
+            # workfile version
+            collected_version = instance.context.data.get("version")
+        if collected_version is not None:
```

Comment:
But I think only this line is necessary, that previous with `or` could stay. Or is it here because it is more explicit/readable?

Reply:
`instance.data.get("version")` would evaluate 0 to a falsey value. As such, `instance.data.get("version") or instance.context.data.get("version")` would use the context one if instance data version was set to 0... which it should not.
So we need an explicit check. Or?

Reply:
You are right, I always thing that part after 'or' will trigger only if first part is None, my bad, please ignore me.